soda_machine.rb

```diff
@@ -1,18 +1,45 @@
+require_relative 'soda'
+
 class SodaMachine
   attr_reader :sodas, :cash
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+   # @new_soda = Soda.new
   end
 
+  # RETURNS current inventory count
   def current_inventory_count
+    return @sodas.length
   end
 
+  # finds a certain soda in the machine
   def find_soda(soda_brand)
+
+    sodas.find do |soda|
+      soda.brand == soda_brand
+    end
+
+   # @sodas.each do |soda|
+   #     return @soda
+   #   else
+   #    return nil
+   #   end
+   # end
   end
 
   def sell(soda_brand)
+    i = @sodas.find_index do |soda|
+      soda.brand == soda_brand
+    end
```

Comment:
Could just reuse the `find_soda` method here.  You'd just need to use `#delete` instead of `#delete_at` below.